pkg/vault/vault_secret_client.go

```diff
@@ -143,22 +148,28 @@ func (v *VaultClient) GetSecret(token, secretPath string) ([]byte, error) {
 	return responseBody, nil
 }
 
-const MalformedSecretResponseFromVault = "malformed secret response from vault"
-const FailedToExtractS3CredentialsFromVaultSecret = "failed to extract S3 credentials from Vault secret"
+func (v *VaultClient) GetSecretMap(credentialsPath *string) (map[string]interface{}, error) {
+	token, err := v.getToken()
+	if err != nil {
```

Comment:
please remind me why do you use "Warn" instead of "Err" 
you mask the real errors. Please add the into the log.

Reply:
The reason that I chose `Warn` is that I can imagine scenarios where the connector is given all necessary credentials, and does not need credentials hidden in a secret.
If you think that is a mistake, I will change it.